### PR TITLE
Fix comparison for partial and full resource urls

### PR DESCRIPTION
**Pull request by panslava**
some partial link do not even have project set...

/assign @spencerhance @swetharepakula

### CODE REVIEW

File: pkg/utils/utils_test.go [hunk 1]
```diff
@@ -179,6 +179,11 @@ func TestEqualResourceIDs(t *testing.T) {
 			b:    "projects/project-id/zones/us-central1-a/instanceGroups/example-group",
 			want: true,
 		},
+		"partial vs full 2": {
```
Comment (swetharepakula) [inline pkg/utils/utils_test.go:182]: nit: indicate partial without project ID

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -393,6 +393,20 @@ func EqualResourceIDs(a, b string) bool {
 		return false
 	}
 
+	if aId.APIGroup == "" {
```
Comment (swetharepakula) [inline pkg/utils/utils.go:396]: update the comment to indicate that if project or API group is missing, that those are ignored

**Review by swetharepakula — COMMENTED** [review]
Generally this looks safe, but we should ensure that this is the expectation at all the call sites. We are adding some unique behavior under the hood of a utils function which may be unexpected. 

There also may be upcoming features where we do need to verify project information

**Comment by swetharepakula** [comment]
There also seems to be other ResourceURLs functions in utils: https://github.com/kubernetes/ingress-gce/blob/5284ec45ac7ddb87497290eed1001e25e34a07c8/pkg/utils/utils.go#L339-L382

Do any of those do the same thing as what you are looking fo?

**Comment by panslava** [comment]
Thank you for review @swetharepakula 

Talked offline, there is a better function that suits this case "EqualResourcePaths"

Decided to use it, and just added unit tests for utils

**Review by swetharepakula — COMMENTED** [review]
/lgtm

**Pull request merged**